src/websocket/websocket_endpoint_base.cpp

```diff
@@ -0,0 +1,29 @@
+#include "websocket_endpoint_base.h"
+#include <memory>
+namespace sensr {
+  WebSocketEndPointBase::WebSocketEndPointBase()
+  : status_(Status::kConnecting), msg_receiver_(0), err_receiver_(0) {}
```

Comment:
Can we call the default constructor of `std::function` instead of 0? or you can even omit them here. 
```suggestion
  : status_(Status::kConnecting), msg_receiver_(), err_receiver_() {}
```